pkg/project/handlers.go

```diff
@@ -14,7 +16,32 @@ import (
 	kclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func SetProjectSupportedRegions(req router.Request, resp router.Response) error {
+// SetDefaultComputeClass sets the default compute class status field of a [v1.ProjectInstance] to the value of its spec
+// field if set.
+func SetDefaultComputeClass(req router.Request, resp router.Response) error {
+	project := req.Object.(*v1.ProjectInstance)
+	if cc := project.Spec.DefaultComputeClass; cc != "" && project.Status.DefaultComputeClass != cc {
+		// The spec has been changed, update the status field to match.
+		project.Status.DefaultComputeClass = cc
+	}
```

Comment:
Similar to default and supported region(s), we should always have a status value for default compute class. Otherwise, the status field is not necessary.